src/page-modules/assistant/server/journey-planner/index.ts

```diff
@@ -0,0 +1,265 @@
+                  }
+                : null,
+            },
+            toPlace: {
+              name: leg.toPlace.name ?? null,
+              longitude: leg.toPlace.longitude,
+              latitude: leg.toPlace.latitude,
+              quay: leg.toPlace.quay
+                ? {
+                    publicCode: leg.toPlace.quay.publicCode ?? null,
+                    name: leg.toPlace.quay.name,
+                    id: leg.toPlace.quay.id,
+                    situations: mapSituations(leg.toPlace.quay.situations),
+                  }
+                : null,
+            },
+            serviceJourney: leg.serviceJourney
+              ? {
+                  id: leg.serviceJourney.id,
+                  notices: mapNotices(leg.serviceJourney.notices),
+                  journeyPattern: leg.serviceJourney.journeyPattern
+                    ? {
+                        notices: mapNotices(
+                          leg.serviceJourney.journeyPattern.notices,
+                        ),
+                      }
+                    : null,
+                }
+              : null,
+            interchangeTo: leg.interchangeTo
+              ? {
+                  guaranteed: leg.interchangeTo?.guaranteed ?? false,
+                  toServiceJourney: leg.interchangeTo.toServiceJourney?.id
+                    ? {
+                        id: leg.interchangeTo.toServiceJourney.id,
+                      }
+                    : null,
+                }
+              : null,
+            pointsOnLink:
+              leg.pointsOnLink?.points && leg.pointsOnLink?.length
+                ? {
+                    points: leg.pointsOnLink.points,
+                    length: leg.pointsOnLink.length,
+                  }
+                : null,
+            intermediateEstimatedCalls: leg.intermediateEstimatedCalls.map(
+              (intermediateEstimatedCall) => ({
+                date: intermediateEstimatedCall.date,
+                quay: {
+                  id: intermediateEstimatedCall.quay.id,
+                  name: intermediateEstimatedCall.quay.name,
+                },
+              }),
+            ),
+            authority: leg.authority?.id
+              ? {
+                  id: leg.authority.id,
+                }
+              : null,
+            serviceJourneyEstimatedCalls: null,
+            datedServiceJourney: null,
+          })),
+        })),
+      };
+
+      const validated = tripSchema.safeParse(data);
+      if (!validated.success) {
+        throw validated.error;
+      }
+
+      return validated.data;
+    },
+  };
+
+  return api;
+}
+
+type RecursivePartial<T> = {
+  [P in keyof T]?: T[P] extends (infer U)[]
+    ? RecursivePartial<U>[]
+    : T[P] extends object | undefined
+    ? RecursivePartial<T[P]>
+    : T[P];
+};
+
+function mapSituations(
+  situations: TripsQuery['trip']['tripPatterns'][0]['legs'][0]['situations'][0][],
```

Comment:
Got no type for `situations`?

Reply:
```ts
export type Situation = z.infer<typeof situationSchema>;
```

This is not the same Situation?

Reply:
The difference is that some of the values in `TripsQuery['trip']['tripPatterns'][0]['legs'][0]['situations'][0][]` can be `undefined` while the same values in `Situation` cannot (`null` instead). I thought about creating a second type, e.g. `GraphQLSituation`, but if the auto-generated type changes we would have to update that one as well. I don't think that is a good approach either 🤔

Reply:
Hmm, okay.

Reply:
What do you think? Is it better to create a new type or do you see an alternative solution to this?

Reply:
I'm not a huge fan of the indexed type, but I see the arguments for not creating a second type  that needs to be in sync. @mikaelbr thoughts?

Reply:
There is a lot of mapping here that is mostly undefined/defined stuff. Feels like this is something that we could do with `zod` because I don't think all of these should actually be nullable? But I think that can also just be noted as a future improvement.